numpy/linalg/linalg.py

```diff
@@ -1010,8 +991,6 @@ def eigvalsh(a, UPLO='L'):
     _assertRankAtLeast2(a)
     _assertNdSquareness(a)
     t, result_t = _commonType(a)
-    if _isEmpty2d(a):
```

Comment:
@eric-wieser can you have a look if you have time. Since I don't know all your newer changes, and I guess it may be that the tests are not quite how they would be nice (dunno how well the rebase worked). Also this moves some of your fixups into the C-lapack wrappers.